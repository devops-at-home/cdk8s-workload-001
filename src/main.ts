import { App } from 'cdk8s';
import { HelloKubernetes } from './charts/hello-kubernetes.chart';
import { CertManagerWebhookExternalDns } from './charts/cert-manager-webhook-externaldns.chart';

const app = new App();

new HelloKubernetes(app, 'Chart-HelloKubernetes');

new CertManagerWebhookExternalDns(app, 'chart-CertManagerWebhookExternalDns', {
    namespace: 'cert-manager',
    domain: process.env.DOMAIN,
});

app.synth();
