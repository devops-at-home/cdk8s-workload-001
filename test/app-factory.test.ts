import { App } from 'cdk8s';
import { appFactory } from '../src/main';

describe('Snapshot test for appFactory chart', () => {
    const app = new App();

    appFactory(app);

    const charts = app.charts.map((chart) => {
        return { name: chart.labels['chart'], manifest: chart };
    });

    test.each(charts)('$name', ({ manifest }) => {
        expect(manifest.toJson()).toMatchSnapshot();
    });
});
