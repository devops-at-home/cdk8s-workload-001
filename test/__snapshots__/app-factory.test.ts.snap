// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for appFactory chart certificate-issuer 1`] = `
[
  {
    "apiVersion": "cert-manager.io/v1",
    "kind": "ClusterIssuer",
    "metadata": {
      "labels": {
        "chart": "certificate-issuer",
      },
      "name": "certificate-issuer-cloudflare-staging-c8a71d40",
      "namespace": "cert-manager",
    },
    "spec": {
      "acme": {
        "privateKeySecretRef": {
          "name": "letsencrypt-staging",
        },
        "server": "https://acme-staging-v02.api.letsencrypt.org/directory",
        "solvers": [
          {
            "dns01": {
              "cloudflare": {
                "apiTokenSecretRef": {
                  "key": "cloudflare_api_token",
                  "name": "cloudflare",
                },
              },
            },
          },
        ],
      },
    },
  },
]
`;

exports[`Snapshot test for appFactory chart hello-kubernetes 1`] = `
[
  {
    "apiVersion": "external-secrets.io/v1alpha1",
    "kind": "SecretStore",
    "metadata": {
      "labels": {
        "chart": "hello-kubernetes",
      },
      "name": "hello-kubernetes-secret-store-c879c760",
      "namespace": "default",
    },
    "spec": {
      "provider": {
        "aws": {
          "auth": {
            "secretRef": {
              "accessKeyIDSecretRef": {
                "key": "access-key",
                "name": "awssm-secret",
              },
              "secretAccessKeySecretRef": {
                "key": "secret-access-key",
                "name": "awssm-secret",
              },
            },
          },
          "region": "ap-southeast-2",
          "service": "ParameterStore",
        },
      },
    },
  },
  {
    "apiVersion": "external-secrets.io/v1alpha1",
    "kind": "ExternalSecret",
    "metadata": {
      "labels": {
        "chart": "hello-kubernetes",
      },
      "name": "hello-kubernetes-secret-c80c0f45",
      "namespace": "default",
    },
    "spec": {
      "data": [
        {
          "remoteRef": {
            "key": "MESSAGE",
          },
          "secretKey": "MESSAGE",
        },
      ],
      "refreshInterval": "999h",
      "secretStoreRef": {
        "kind": "SecretStore",
        "name": "hello-kubernetes-secret-store-c879c760",
      },
      "target": {},
    },
  },
  {
    "apiVersion": "k8s.cni.cncf.io/v1",
    "kind": "NetworkAttachmentDefinition",
    "metadata": {
      "labels": {
        "chart": "hello-kubernetes",
      },
      "name": "hello-kubernetes-networkattachmentdefinition-c858201d",
      "namespace": "default",
    },
    "spec": {
      "config": "",
    },
  },
  {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": {
      "labels": {
        "chart": "hello-kubernetes",
      },
      "name": "hello-kubernetes-service-account-c84aaf83",
      "namespace": "default",
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "labels": {
        "chart": "hello-kubernetes",
      },
      "name": "hello-kubernetes-deployment-c8f758bb",
      "namespace": "default",
    },
    "spec": {
      "replicas": 1,
      "selector": {
        "matchLabels": {
          "chart": "hello-kubernetes",
        },
      },
      "template": {
        "spec": {
          "containers": [
            {
              "env": [
                {
                  "name": "PORT",
                  "value": "8080",
                },
                {
                  "name": "MESSAGE",
                  "valueFrom": {
                    "secretKeyRef": {
                      "key": "MESSAGE",
                      "name": "hello-kubernetes-secret-c80c0f45",
                    },
                  },
                },
                {
                  "name": "KUBERNETES_NAMESPACE",
                  "valueFrom": {
                    "fieldRef": {
                      "fieldPath": "metadata.namespace",
                    },
                  },
                },
                {
                  "name": "KUBERNETES_POD_NAME",
                  "valueFrom": {
                    "fieldRef": {
                      "fieldPath": "metadata.name",
                    },
                  },
                },
                {
                  "name": "KUBERNETES_NODE_NAME",
                  "valueFrom": {
                    "fieldRef": {
                      "fieldPath": "spec.nodeName",
                    },
                  },
                },
                {
                  "name": "CONTAINER_IMAGE",
                  "value": "paulbouwer/hello-kubernetes:1.10.1",
                },
              ],
              "image": "paulbouwer/hello-kubernetes:1.10.1",
              "imagePullPolicy": "IfNotPresent",
              "livenessProbe": {
                "httpGet": {
                  "port": 8080,
                },
              },
              "name": "hello-kubernetes",
              "ports": [
                {
                  "containerPort": 8080,
                  "name": "http",
                  "protocol": "TCP",
                },
              ],
              "readinessProbe": {
                "httpGet": {
                  "port": 8080,
                },
              },
            },
          ],
          "serviceAccountName": "hello-kubernetes-service-account-c84aaf83",
        },
      },
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "labels": {
        "chart": "hello-kubernetes",
      },
      "name": "hello-kubernetes-service-c8d4454f",
      "namespace": "default",
    },
    "spec": {
      "ports": [
        {
          "name": "http",
          "port": 8080,
        },
      ],
      "selector": {
        "chart": "hello-kubernetes",
      },
      "type": "LoadBalancer",
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Secret",
    "metadata": {
      "labels": {
        "chart": "hello-kubernetes",
      },
      "name": "hello-kubernetes-secret-tls-c88b494f",
      "namespace": "default",
    },
  },
  {
    "apiVersion": "networking.k8s.io/v1",
    "kind": "Ingress",
    "metadata": {
      "annotations": {
        "cert-manager.io/cluster-issuer": "certificate-issuer-cloudflare-staging-c8a71d40",
        "kubernetes.io/ingress.class": "traefik",
        "traefik.ingress.kubernetes.io/router.tls": "true",
      },
      "labels": {
        "chart": "hello-kubernetes",
      },
      "name": "hello-kubernetes-ingress-c8762254",
      "namespace": "default",
    },
    "spec": {
      "rules": [
        {
          "host": "test.h6020-001.devops-at-ho.me",
          "http": {
            "paths": [
              {
                "backend": {
                  "service": {
                    "name": "hello-kubernetes-service-c8d4454f",
                    "port": {
                      "number": 8080,
                    },
                  },
                },
                "path": "/",
                "pathType": "Prefix",
              },
            ],
          },
        },
      ],
      "tls": [
        {
          "hosts": [
            "test.h6020-001.devops-at-ho.me",
          ],
          "secretName": "hello-kubernetes-secret-tls-c88b494f",
        },
      ],
    },
  },
]
`;

exports[`Snapshot test for appFactory chart metal-lb-config 1`] = `
[
  {
    "apiVersion": "metallb.io/v1beta1",
    "kind": "IPAddressPool",
    "metadata": {
      "labels": {
        "chart": "metal-lb-config",
      },
      "name": "metal-lb-config-ipaddresspool-c8fe577c",
      "namespace": "kube-system",
    },
    "spec": {
      "addresses": [
        undefined,
      ],
    },
  },
  {
    "apiVersion": "metallb.io/v1beta1",
    "kind": "L2Advertisement",
    "metadata": {
      "labels": {
        "chart": "metal-lb-config",
      },
      "name": "metal-lb-config-l2advertisement-c8c28aa3",
      "namespace": "kube-system",
    },
    "spec": {
      "interfaces": [
        "tailscale0",
      ],
      "ipAddressPools": [
        "metal-lb-config-ipaddresspool-c8fe577c",
      ],
    },
  },
]
`;
